src/main/java/edu/ksu/canvas/attendance/submitter/AssignmentValidator.java

```diff
@@ -0,0 +1,96 @@
+package edu.ksu.canvas.attendance.submitter;
+
+
+import edu.ksu.canvas.attendance.entity.AttendanceAssignment;
+import edu.ksu.canvas.attendance.exception.CanvasOutOfSyncException;
+import edu.ksu.canvas.attendance.model.AttendanceSummaryModel;
+import edu.ksu.canvas.attendance.services.CanvasApiWrapperService;
+import edu.ksu.canvas.model.assignment.Assignment;
+import edu.ksu.canvas.oauth.OauthToken;
+import org.apache.log4j.Logger;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+import java.util.Optional;
+
+@Component
+public class AssignmentValidator {
+
+    private static final Logger LOG = Logger.getLogger(AssignmentValidator.class);
+
+    public void validateAttendanceAssignment(Long courseId, AttendanceAssignment attendanceAssignment, CanvasApiWrapperService canvasApiWrapperService, OauthToken oauthToken) throws IOException, CanvasOutOfSyncException {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this method to throw at most one checked exception instead of: java.io.IOException, edu.ksu.canvas.attendance.exception.CanvasOutOfSyncException [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.ome.ksu.edu/coding_rules#rule_key=squid%3AS1160)